src/ucs/sys/sock.c

```diff
@@ -128,9 +129,19 @@ const char *ucs_socket_getname_str(int fd, char *str, size_t max_size)
 static ucs_status_t ucs_socket_check_errno(int io_errno)
 {
     if ((io_errno == EAGAIN) || (io_errno == EWOULDBLOCK) || (io_errno == EINTR)) {
+        /* IO operation or connection establishment procedure was interrupted
+         * or would block and need to try again */
         return UCS_ERR_NO_PROGRESS;
     }
 
+    if ((io_errno == ECONNRESET) || (io_errno == ECONNREFUSED)) {
+        /* Connection reset by peer */
+        return UCS_ERR_NOT_CONNECTED;
```

Comment:
does it make sense to use UCS_ERR_CONNECTION_RESET here?

Reply:
do you mean `UCS_ERR_REJECTED`?

Reply:
actually we can use both

Reply:
IIRC at first iteration of this code we did not have `UCS_ERR_CONNECTION_RESET`